viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
+# I feel like there's a better way to test if title, genre, & rating are truthy 
+# but not sure how
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    if title and genre and rating:
+        return new_movie
+    else:
+        return None
+
+
+def add_to_watched(user_data, movie):
+    has_watched = [movie] 
+    user_data = {"watched" : has_watched }
+    return user_data
+
 
+def add_to_watchlist(user_data, movie):
+    wants_to_watch = [] 
+    user_data = {"watchlist" : wants_to_watch }
+    wants_to_watch.append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title:
+            user_data["watchlist"].remove(movies)
+            user_data["watched"].append(movies)
+    return user_data
+        
+        
 # -----------------------------------------
-# ------------- WAVE 2 --------------------
 # -----------------------------------------
+# ------------- WAVE 2 --------------------
+# -----------------------------------------        
+
+def get_watched_avg_rating(user_data):
+    count = 0
+    movie_rating = 0.0
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        for movies in user_data["watched"]:
+            movie_rating += movies["rating"]
+            count += 1  
+        return movie_rating/count
+
+
 
+def get_most_watched_genre(user_data):
+    genre_list = []
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            genre_list.append(movie["genre"])
+        return max(set(genre_list), key = genre_list.count)

```

Comment:
Same comment as above, you can write `if not user_data["watched"]:` and then un-indent the logic on lines 58-60 (when you get rid of the else block).

Also a note about `max`. Nice use of max here. I find it useful to also practice writing your own implementation of finding a max value from a dictionary. You may find in an interview if you use built in functions like max that an interviewer may ask you how the function works under the hood. So knowing how to write it on your own can help you better understand the function.